src/BlockingDriver.php

```diff
@@ -322,15 +187,21 @@ public function scandir(string $path): Promise
      */
     public function chmod(string $path, int $mode): Promise
     {
-        return new Success((bool) @\chmod($path, $mode));
+        if (!@\chmod($path, $mode)) {
+            return new Failure(new FilesystemException("Could not change file permissions"));
+        }
+
+        StatCache::clear();
+
+        return new Success();
     }
 
     /**
      * {@inheritdoc}
      */
-    public function chown(string $path, int $uid, int $gid): Promise
+    public function chown(string $path, ?int $uid, ?int $gid = null): Promise
```

Comment:
Should we have a default for `$gid` at all?

Reply:
Yeah, I was considering that too... I kinda think we should keep the default in the chown function but drop it in DriverInterface. Do you agree?

Reply:
Yes, seems fine.

Reply:
I applied the same treatment to the null defaults of `Driver::touch()` as well. Not sure if I should do it for `Driver::mkdir()` too...